### ISSUE TITLE
Upgrade package to use RestSharp v107+

### ISSUE THREAD

**Issue by gregkalbach** [okta/okta-sdk-dotnet#606]
### Describe the feature request?

We are attempting to update our application from using okta’s .Net SDK v5x to v6.0.1.
In doing so, we run into versioning errors with the RestSharp library now used by the v6 SDK.
The SDK uses v106 of the RestSharp library. RestSharp introduced breaking changes in v107, which was released in January of 2022. Because the SDK uses v106, and we have other packages that use later versions of RestSharp (108) - packages which we can’t downgrade - we cannot use the v6 SDK.

Request that the okta .net sdk be updated to use a more recent version of the RestSharp library.

### New or Affected Resource(s)

Would affect all endpoints in the SDK.

### Provide a documentation link

https://restsharp.dev/v107/#restsharp-v107

**Comment by laura-rodriguez** [comment]
Hi @gregkalbach,

Thanks for your request. We're not planning to upgrade RestSharp soon, but we'll monitor this issue to see if there's enough interest from the community.

**Comment by dsbegnoche** [comment]
+1. 
When upgrading to v6.0.2, RestSharp was also upgraded to 108.0.1, causing the following exception.

```
Unhandled exception. System.MissingMethodException: Method not found: 'Void RestSharp.RestRequest..ctor(RestSharp.Method)'.
   at Okta.Sdk.Client.ApiClient.NewRequest(HttpMethod method, String path, RequestOptions options, IReadableConfiguration configuration)
   at Okta.Sdk.Client.ApiClient.GetAsync[T](String path, RequestOptions options, IReadableConfiguration configuration, CancellationToken cancellationToken)
```
   
 We are able to fix by directly installing RestSharp 106.0.13 in our solution. 

 IMO, if Okta is not upgrading to the latest version or RestSharp, the Nuget dependency should be locked down to 106.13.0 to help others avoid this issue.

**Comment by rcollette** [comment]
Shouldn't this issue be closed?

**Comment by laura-rodriguez** [comment]
Hey @rcollette,

We haven't closed this issue yet, because we're still working on the release 7.0.0. To learn about what's coming on 7.0.0 (which should be happening soon) you can check out this [PR](https://github.com/okta/okta-sdk-dotnet/pull/654)

**Comment by laura-rodriguez** [comment]
Fixed in 7.0.0
